tsdb/points.go

```diff
@@ -987,7 +996,8 @@ type Fields map[string]interface{}
 
 func parseNumber(val []byte) (interface{}, error) {
 	for i := 0; i < len(val); i++ {
-		if val[i] == '.' {
+		// If there is a decimal or an N (NaN), parse as float
+		if val[i] == '.' || val[i] == 'N' || val[i] == 'n' || val[i] == 'I' || val[i] == 'i' {
```

Comment:
Ok. It's a bit obscure, so a comment or const-out the magic character would
be good.

On Tuesday, July 7, 2015, Jason Wilder <notifications@github.com> wrote:

> In tsdb/points.go
> <https://github.com/influxdb/influxdb/pull/3248#discussion_r34054680>:
>
> > @@ -987,7 +996,8 @@ type Fields map[string]interface{}
> >
> >  func parseNumber(val []byte) (interface{}, error) {
> >  	for i := 0; i < len(val); i++ {
> > -		if val[i] == '.' {
> > +		// If there is a decimal or an N (NaN), parse as float
> > +		if val[i] == '.' || val[i] == 'N' || val[i] == 'n' || val[i] == 'I' || val[i] == 'i' {
>
> Inf. It's a valid value for strconv.ParseFloat. It was causing a panic
> previous because it would fall-through to the boolean parsing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/pull/3248/files#r34054680>.
>